src/main/java/io/debezium/connector/cassandra/SchemaHolder.java

```diff
@@ -5,154 +5,305 @@
 
-    public synchronized void removeTableSchema(KeyspaceTable kst) {
+    private void removeTableSchema(KeyspaceTable kst) {
         tableToKVSchemaMap.remove(kst);
+        LOGGER.info("Removed the schema for {}.{} from table schema cache.", kst.keyspace, kst.table);
     }
 
-    // there is not "addKeyspace", it is not necessary
-    // as we will ever add a concrete table (with keyspace) but we will also dropping all tables when keyspace is dropped
-    public synchronized void addOrUpdateTableSchema(KeyspaceTable kst, KeyValueSchema kvs) {
+    private void addOrUpdateTableSchema(KeyspaceTable kst, KeyValueSchema kvs) {
+        boolean isUpdate = tableToKVSchemaMap.containsKey(kst);
         tableToKVSchemaMap.put(kst, kvs);
+        if (isUpdate) {
+            LOGGER.info("Updated the schema for {}.{} in table schema cache.", kst.keyspace, kst.table);
+        }
+        else {
+            LOGGER.info("Added the schema for {}.{} to table schema cache.", kst.keyspace, kst.table);
+        }
     }
 
-    /**
-     * Get the schema of an inner field based on the field name
-     * @param fieldName the name of the field in the schema
-     * @param schema    the schema where the field resides in
-     * @return Schema
-     */
-    public static Schema getFieldSchema(String fieldName, Schema schema) {
-        if (schema.type().equals(Schema.Type.STRUCT)) {
-            return schema.field(fieldName).schema();
-        }
-        throw new CassandraConnectorSchemaException("Only STRUCT type is supported for this method, but encountered " + schema.type());
-    }
-
-    public static class KeyValueSchema {
-        private final TableMetadata tableMetadata;
-        private final Schema keySchema;
-        private final Schema valueSchema;
+    class CassandraSchemaChangeListener implements SchemaChangeListener {
 
-        KeyValueSchema(String kafkaTopicPrefix, TableMetadata tableMetadata, SourceInfoStructMaker sourceInfoStructMaker) {
-            this.tableMetadata = tableMetadata;
-            this.keySchema = getKeySchema(kafkaTopicPrefix, tableMetadata);
-            this.valueSchema = getValueSchema(kafkaTopicPrefix, tableMetadata, sourceInfoStructMaker);
+        @Override
+        public void onKeyspaceAdded(final KeyspaceMetadata keyspaceMetadata) {
+            try {
+                Schema.instance.setKeyspaceMetadata(org.apache.cassandra.schema.KeyspaceMetadata.create(
+                        keyspaceMetadata.getName(),
+                        KeyspaceParams.create(keyspaceMetadata.isDurableWrites(),
+                                keyspaceMetadata.getReplication())));
+                Keyspace.openWithoutSSTables(keyspaceMetadata.getName());
+                LOGGER.info("Added keyspace [{}] to Schema Instance.", keyspaceMetadata.asCQLQuery());
+            }
+            catch (Exception e) {
+                LOGGER.warn("Error happened while adding the keyspace {} to Schema instance", keyspaceMetadata.getName(), e);
```

Comment:
```suggestion
                LOGGER.warn("Error happened while adding the keyspace {} to schema instance", keyspaceMetadata.getName(), e);
```

So should we truly swallow this? Or rather bubble up? I.e. should the connector actually keep going? What will happen later on when parsing a change event for that key space?

Reply:
Again this is not related to this PR, I was just copying the codes from SchemaProcessor in PR#26. If Cassandra Connector stops, it will potentially cause write suspension in Cassandra DB. For this reason, we want to avoid throwing exceptions in this connector as much as possible.

Reply:
> So should we truly swallow this? Or rather bubble up? I.e. should the connector actually keep going? What will happen later on when parsing a change event for that key space?

Again this is not related to this PR, I was just copying the codes from SchemaProcessor in PR#26. If Cassandra Connector stops, it will potentially cause write suspension in Cassandra DB. For this reason, we want to avoid throwing exceptions in this connector as much as possible.